### ISSUE TITLE
Feature Request: SendGrid Integration for Replies via Email

### ISSUE THREAD

**Issue by raamdev** [websharks/comment-mail#200]
http://sendgrid.com/

Comment Mail currently integrates with Mandrill, however Mandrill charges $24.95 if you have 100,000 emails per month whereas SendGrid has a pay-as-you-use tier which costs $0.10 per 1,000 emails, so it's only $10 for 100,000 emails.

**Comment by jaswsinc** [comment]
As long as you are only using Mandrill as an RVE handler, it is 100% free.
http://comment-mail.com/kb-article/mandrill-rve-handler/#toc-e8059774
https://mandrill.zendesk.com/hc/en-us/articles/205583197-Inbound-Email-Processing-Overview

If you use them as an SMTP provider, then it costs you.



> $0.10 per 1,000 emails

I think it's the same with AWS SES.

**Comment by raamdev** [comment]
[Mandrill writes...](http://mandrill.com/pricing/)
> Once you’ve finished your free trial, it’s $9.95/month 

[SendGrid writes...](https://sendgrid.com/pricing)
> Pay as you go with SendGrid Lite $0.10 / 1,000 emails.

So with Mandrill you pay a minimum of $9.95/month (even if all you have is inbound email), whereas with SendGrid you can have a truly free account without a time restriction.

**Comment by raamdev** [comment]
> with Mandrill you pay a minimum of $9.95/month

Actually, now that I've created an account I see that's probably not the case. It looks like the 30 days "free trial" is just referring to access to support:

![2016-01-09_07-34-40](https://cloud.githubusercontent.com/assets/53005/12216069/935ce5de-b6a3-11e5-8977-80fe40b9b25a.png)

**Comment by raamdev** [comment]
In any case, I still think it's worth exploring SendGrid integration for RVE, as many site owners may already be using SendGrid for their site (it's a _lot_ cheaper than Mandrill for outgoing email--you need to send 6 million emails a month before you hit the $0.10/1000 pricing that SendGrid offers for their free account).

**Comment by jaswsinc** [comment]
> Actually, now that I've created an account I see that's probably not the case. It looks like the 30 days "free trial" is just referring to access to support

Yeah, I think so too. I've had two accounts with Mandrill for over 2 years now and I have never been charged for the Inbound Mail Routes feature. I use that feature for several things, because it is an easy way to pipe data from other services to email, and then to a webhook with structured data that you can parse in PHP scripts or otherwise.

> In any case, I still think it's worth exploring SendGrid integration for RVE, as many site owners may already be using SendGrid for their site (it's a lot cheaper than Mandrill for outgoing email--you need to send 6 million emails a month before you hit the $0.10/1000 pricing that SendGrid offers for their free account).

I think we can already say that we integrate with SendGrid, because CM allows you to integrate with _any_ SMTP provider. So if you have a SendGrid account you can put your SendGrid SMTP access credentials into Comment Mail and that's how email notifications will be sent.

I agree it would be nice if CM integrated with SendGrid's Inbound handlers too, but I haven't tested those myself yet. Do they offer that service? It looks like they do. https://sendgrid.com/docs/API_Reference/Webhooks/parse.html

**Comment by raamdev** [comment]
Yep, it looks like they do, however there's this big note near the top of that page:

> Available to all accounts except Lite.

So, if you're looking for the cheapest option, it looks like the Mandrill still wins here, but for site owners who already use SendGrid it would be nice to be able to enable RVE via SendGrid too.

**Comment by jaswsinc** [comment]
You beat me to to it! :-)

---

I noticed on this page it says...
https://sendgrid.com/docs/API_Reference/Webhooks/parse.html

> Available to all accounts except Lite.

And on this page (for the Lite pricing) it says...
https://sendgrid.com/pricing?mc=SendGrid%20Documentation

> The Lite plan excludes the following features: advanced reporting and analytics, Event Webhook, Parse Webhook, Subuser API, DKIM customization, dedicated IPs, IP whitelabeling, IP warm up, feedback loops, Marketing Campaigns.

**Comment by jaswsinc** [comment]
> IP warm up

haha... Lite users get no IP foreplay action I guess.

**Comment by raamdev** [comment]
Also mentioned here (private ticket): https://websharks.zendesk.com/agent/tickets/10680

**Comment by kristineds** [comment]
:+1:

**Comment by raamdev** [comment]
Related: https://github.com/websharks/comment-mail/issues/265
